libs/remix-debug/src/code/codeManager.ts

```diff
@@ -145,15 +145,36 @@ export class CodeManager {
     })
   }
 
-  private retrieveIndexAndTrigger (codeMananger, address, step, code) {
+  private async retrieveIndexAndTrigger (codeMananger, address, step, code) {
     let result
+    let next
+    const returnInstructionIndexes = []
+    const outOfGasInstructionIndexes = []
+
     try {
       result = codeMananger.getInstructionIndex(address, step)
+      next = codeMananger.getInstructionIndex(address, step + 1)
```

Comment:
I think line 156 to 164 can come in a single loop.

Reply:
done